bench/workgen/workgen.cxx

```diff
@@ -1042,12 +1044,24 @@ ThreadRunner::op_run(Operation *op)
     // be retried.
     if (op->is_table_op()) {
         op->kv_gen(this, true, 100, recno, _keybuf);
-        cursor->set_key(cursor, _keybuf);
+        if (std::string(cursor->key_format) == "u") {
+            item.data = _keybuf;
+            item.size = strlen(_keybuf);
+            cursor->set_key(cursor, &item);
+        } else {
```

Comment:
Now that we're checking the `key_format`, let's confirm that (if isn't `u`), then it must be `S`.  Otherwise, do a
```
            THROW("key_format must be ....");
```
Same with `value_format`.

Reply:
I used an `ASSERT` instead, let me know if that's ok. See ee1f260370b40ed0e8f1c07abc963c19a94dcdf9.

Reply:
@etienneptl It's okay, I think I'd prefer a throw for a user configuration error.

Reply:
No worries, I replaced the `assert` `with a `throw` in d08cdd3f79cae992e49fef822235429d6b88cb89.